src/java.desktop/share/classes/com/sun/imageio/plugins/bmp/BMPImageWriter.java

```diff
@@ -1456,6 +1456,9 @@ protected boolean canEncodeImage(int compression, ImageTypeSpecifier imgType) {
         }
         int biType = imgType.getBufferedImageType();
         int bpp = imgType.getColorModel().getPixelSize();
+        if (bpp != 0 && bpp != 1 && bpp != 4 && bpp != 8 && bpp != 16 && bpp != 24 && bpp != 32) {
```

Comment:
This change looks good to me.

Reply:
Probably the exception message caused by this should be updated as well? Currently, it takes care of compression only.

Reply:
I also think the message should be change. I added bpp information at the end of the exception message.